pymongo/mongo_client.py

```diff
@@ -368,45 +369,57 @@ def __init__(
-            version in use does not support loading system CA certificates
-            then the ``ssl_ca_certs`` parameter must point to a file of CA
-            certificates. Implies ``ssl=True``. Defaults to
-            ``ssl.CERT_REQUIRED`` if not provided and ``ssl=True``.
-          - `ssl_ca_certs`: The ca_certs file contains a set of concatenated
+          - `tls`: (boolean) If ``True``, create the connection to the server
+            using TLS. Defaults to ``False``.
+          - `ssl`: (boolean) Alias for ``tls``.
+          - `tlsInsecure`: (boolean) Specify whether TLS constraints should be
+            relaxed as much as possible. Setting ``tlsInsecure=True`` implies
+            ``tlsAllowInvalidCertificates=False`` and
+            ``tlsAllowInvalidHostnames=False``. Defaults to ``False``. Think
+            very carefully before setting this to ``True`` as it could allow
+            malicious third-parties to exploit TLS vulnerabilities to attack
+            your application.
+          - `tlsAllowInvalidCertificates`: (boolean) Specifies whether a
+            certificate is required from the other side of the connection, and
+            whether it will be validated if provided. Certificates are ignored
+            if ``tlsAllowInvalidCertificates=True``. Certificates are required
+            and validated if ``tlsAllowInvalidCertificates=False``. If the
+            value of this parameter is ``False`` and a value is not provided
+            for ``tlsCAFile``, PyMongo will attempt to load system provided CA
+            certificates. If the python version in use does not support loading
+            system CA certificates then the ``tlsCAFile`` parameter must point
+            to a file of CA certificates. ``tlsAllowInvalidCertificates=False``
+            implies ``tls=True``. Defaults to ``False``.
+          - `tlsAllowInvalidHostnames`: (boolean) If ``False`` and
+            `tlsAllowInvalidCertificates` is ``False``, enables hostname
+            verification using the :func:`~ssl.match_hostname` function from
+            python's :mod:`~ssl` module. ``tlsAllowInvalidHostnames=False``
+            implies ``tls=True``. Defaults to ``False``. Think very carefully
+            before setting this to ``True`` as that could make your application
+            vulnerable to man-in-the-middle attacks.
+          - `tlsCAFile`: The file containing a single or a bundle of
             "certification authority" certificates, which are used to validate
             certificates passed from the other end of the connection.
-            Implies ``ssl=True``. Defaults to ``None``.
+            Implies ``tls=True``. Defaults to ``None``.
+          - `tlsCertificateKeyFile`: The client certificate file or the client
+            private key file used to identify the local connection against
+            mongod. In the case that they are both needed, the files can either
```

Comment:
Both are always needed. We just provide the option of passing them in two separate files. Read this for a more in depth explanation: https://docs.python.org/2/library/ssl.html#certificates

I think the terse original documentation for ssl_certfile and ssl_keyfile is correct.